apps/scorecard/server/dataFunctions/processScorecard.js

```diff
@@ -572,21 +616,24 @@ processScorecard = function (plotParams, plotFunction) {
       cbScorecardPool.upsertCB(id, doc);
     })(id, scDoc).then(() => {
       console.log("upserted doc with id", id);
+      // we do need an actual pause here (500 ms), because the upsert function
```

Comment:
I'm not sure that this is the best way to fix this problem. The problem might not be that the upsert didn't finish, more likely the indexes are not yet consistent. See https://www.couchbase.com/blog/defining-scan-consistency-node-js-couchbase-application/
for an explanation. The call that the status page in matsMethods._getScorecardInfo is ``` SELECT
            sc.id,
            sc.userName,
            sc.name,
            sc.status,
            sc.processedAt as processedAt,
            sc.submitted,
            sc.dateRange
            From
            vxdata._default.SCORECARD sc
            WHERE
            sc.type='SC';``` and it probably needs to be ``` var statement = N1qlQuery.fromString(SELECT
            sc.id,
            sc.userName,
            sc.name,
            sc.status,
            sc.processedAt as processedAt,
            sc.submitted,
            sc.dateRange
            From
            vxdata._default.SCORECARD sc
            WHERE
            sc.type='SC');statement.consistency(N1qlQuery.Consistency.REQUEST_PLUS); ``` - or something like that - in order to add the scanConsistency to make the select wait until the indexes are finished updating.

Reply:
I did try this, and it still needs the pause. It looks like documents are just not immediately available after upsertion.